sql: validate trigger backreferences in table descriptors

spilchen opened the pull request:
Previously, descriptor validation only accounted for backreferences from sequences and views. However, the introduction of triggers introduces table-to-table references, which were not being validated. This change extends the validation logic to cover these cases, improving correctness and robustness of descriptor validation involving triggers.

Fixes: #148167
Epic: CRDB-42942

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149621)
<!-- Reviewable:end -->

spilchen committed:
sql: validate trigger backreferences in table descriptors

Previously, descriptor validation only accounted for backreferences from
sequences and views. However, the introduction of triggers introduces
table-to-table references, which were not being validated. This change extends
the validation logic to cover these cases.

Fixes: #148167
Epic: CRDB-42942

Release note: None

rafiss reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/149621#-:-OVK2zda0RmsV1_B1eV3:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @fqazi)



<!-- Sent from Reviewable.io -->
